repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -474,24 +474,20 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
           mapper.getFactory.configure(JsonGenerator.Feature.ESCAPE_NON_ASCII, true)
           Option(new ByteArrayInputStream(mapper.writeValueAsBytes(destination)))
         } catch {
-          case jpe: JsonProcessingException =>
-            throw SamlPolicyException(SC_BAD_GATEWAY, "Origin service provided bad response", jpe)
+          case e@(_: AttributeMapperException | _: JsonProcessingException) =>
+            throw SamlPolicyException(SC_BAD_GATEWAY, "Origin service provided bad response", e)
         }
       case xml if xml.contains("xml") =>
-        var docBuilder: Option[DocumentBuilder] = None
         try {
-          docBuilder = Option(XMLParserPool.borrowParser)
           val destination = AttributeMapper.addExtendedAttributes(
             new StreamSource(response.getOutputStreamAsInputStream),
             translatedSamlResponse,
             validate = true,
             XSDEngine.AUTO.toString)
           Option(convertDocumentToStream(destination))
         } catch {
-          case se@(_: SAXException | _: SaxonApiException) =>
-            throw SamlPolicyException(SC_BAD_GATEWAY, "Origin service provided bad response", se)
-        } finally {
-          docBuilder.foreach(XMLParserPool.returnParser)
+          case ame: AttributeMapperException =>
```

Comment:
This is another change that does not quite map one-to-one. `AttributeMapperException` is broader than either `SAXException` or `SaxonApiException`, however, I did not want to make things complicated by checking the causes everywhere. Since the intent of this error handling seemed to be to fail when Attribute Mapper fails, I just caught the general exception.